src/lasp_dependence_dag.erl

```diff
@@ -176,6 +256,25 @@ handle_call({export_dot, Path}, _From, #state{dag=Dag}=State) ->
 handle_call(get_process_map, _From, #state{process_map=PM}=State) ->
     {reply, {ok, dict:to_list(PM)}, State};
 
+handle_call(contract, _From, #state{dag=Dag}=State) ->
+    lists:foreach(fun(P) ->
+        contract(Dag, P)
+    end, contraction_paths(Dag)),
+    {reply, ok, State};
+
+handle_call({cleave, Vertex}, _From, #state{dag=Dag, optimized_map=OptMap}=State) ->
+    cleave_if_contracted(Dag, Vertex, OptMap),
+    {reply, ok, State};
+
+handle_call(cleave_all, _From, #state{optimized_map=OptMap}=State) ->
+    lists:foreach(fun(Id) ->
+        {Pid, _, _} = dict:fetch(Id, OptMap),
+        spawn_link(fun() ->
+            lasp_process_sup:terminate_child(lasp_process_sup, Pid)
```

Comment:
To undo the optimization for a given path, it kills the process that connects its endpoints.

When any of those processes are killed by the superviros, the cleaving process is automatically started.

You can see the handler for that in [`handle_info`](https://github.com/ergl/lasp/blob/contraction/src/lasp_dependence_dag.erl#L426) and the actual cleaving in [`cleave_associated_path`](https://github.com/ergl/lasp/blob/contraction/src/lasp_dependence_dag.erl#L832).